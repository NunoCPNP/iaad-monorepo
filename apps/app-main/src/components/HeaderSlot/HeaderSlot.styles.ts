import styled from '@emotion/styled'

export const Container = styled.div`
  background-color: ${({ theme }) => theme.colors.primary};
  color: ${({ theme }) => theme.colors.secondary};

  div {
    display: flex;
    justify-content: space-between;
    max-width: 140rem;

    @media only screen and (max-width: 1400px) {
      padding: 0 1rem;
    }
  }
`

export const Slot = styled.div`
  text-transform: uppercase;
  font-size: 1.2rem;
  display: flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
`

export const Bold = styled.span`
  font-weight: 700;
`

export const Separator = styled.div`
  padding: 0 0.7rem 0 0.7rem;
`

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
`
