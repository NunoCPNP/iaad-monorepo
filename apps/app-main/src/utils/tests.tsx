import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@emotion/react'

import { dark } from '../styles/themes/dark'

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={dark}>{children}</ThemeProvider>
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
