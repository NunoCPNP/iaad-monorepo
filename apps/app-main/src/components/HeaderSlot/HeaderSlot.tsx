import useTranslation from 'next-translate/useTranslation'

import LanguageSelector from '../LanguageSelector'

import { Container, Wrapper, Slot, Bold, Separator } from './HeaderSlot.styles'

const HeaderSlot: React.FC = () => {
  const { t } = useTranslation()

  return (
    <Container data-testid="headerslot">
      <div>
        <div></div>
        <Slot>
          <Bold>{t('common:headerslot.1')}</Bold>
          <Separator>|</Separator>
          <span>{t('common:headerslot.2')}</span>
        </Slot>
        <Wrapper>
          <LanguageSelector />
        </Wrapper>
      </div>
    </Container>
  )
}

export default HeaderSlot
