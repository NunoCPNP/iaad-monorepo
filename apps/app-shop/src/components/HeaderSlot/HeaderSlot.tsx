import { Container, Bold, Separator } from './HeaderSlot.styles'

const HeaderSlot: React.FC = () => (
  <Container data-testid="headerslot">
    <Bold>Stay Safe, Stay Home</Bold>
    <Separator>|</Separator>
    <span>Covid 19</span>
  </Container>
)

export default HeaderSlot
