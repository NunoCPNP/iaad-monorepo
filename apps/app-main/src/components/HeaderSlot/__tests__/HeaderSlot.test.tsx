import { render, screen, cleanup } from '../../../utils/tests'

import HeaderSlot from '../HeaderSlot'

afterEach(cleanup)

describe('<HeaderSlot /> spec', () => {
  it('Should run without errors', () => {
    const { queryAllByTestId } = render(<HeaderSlot />)

    const headerslot = queryAllByTestId(/headerslot/i)
    expect(headerslot).toHaveLength(1)
  })

  it('Should recieve translation strings', () => {
    const { getByText } = render(<HeaderSlot />)

    expect(screen.getByText('common:headerslot.1')).toBeTruthy()
    expect(screen.getByText('common:headerslot.2')).toBeTruthy()
  })

  it('Should render LanguageSelector component', () => {
    const { queryAllByTestId } = render(<HeaderSlot />)

    const headerslot = queryAllByTestId(/languageselector/i)
    expect(headerslot).toHaveLength(1)
  })

  it('Should match snapshot', () => {
    const { asFragment } = render(<HeaderSlot />)

    expect(asFragment(<HeaderSlot />)).toMatchSnapshot()
  })
})
