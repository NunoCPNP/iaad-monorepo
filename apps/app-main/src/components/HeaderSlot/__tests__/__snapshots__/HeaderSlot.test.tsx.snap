// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderSlot /> spec Should match snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #272727;
  color: #FCFAFA;
  text-transform: uppercase;
  font-size: 1.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  height: 3rem;
}

.emotion-1 {
  font-weight: 700;
}

.emotion-2 {
  padding: 0 0.7rem 0 0.7rem;
}

<div
    class="emotion-0"
    data-testid="headerslot"
  >
    <span
      class="emotion-1"
    >
      common:headerslot.1
    </span>
    <div
      class="emotion-2"
    >
      |
    </div>
    <span>
      common:headerslot.2
    </span>
  </div>
</DocumentFragment>
`;
